### PR TITLE
[PM-3763] remove Sweet Alert from desktop and browser

### PR DESCRIPTION
**Pull request by willmartian**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

This PR removes references to Sweet Alert from **browser and desktop clients**. 

Related PRs:

- [Removes references in web](https://github.com/bitwarden/clients/pull/6170)
- [Removes dependency](https://github.com/bitwarden/clients/pull/6191)

## Code changes

Each call to `Swal.fire` was replaced with a new component utilizing `bit-simple-dialog` and call to `DialogService.open`. 

`DialogService.openSimpleDialog` was not used due to the dialogs requiring unsupported formatting, like line breaks or strong text.

New components:

- apps/browser/src/popup/components/desktop-sync-verification-dialog.component.ts
- apps/browser/src/popup/settings/await-desktop-dialog.component.ts
- apps/browser/src/popup/settings/fingerprint-dialog.component.ts
  - I moved this to `@bitwarden/auth` since it is shared between desktop and browser
- apps/desktop/src/app/components/browser-sync-verification-dialog.component.ts
- apps/desktop/src/app/components/verify-native-messaging-dialog.component.ts

Other changes: 

- `libs/components/src/dialog/directives/dialog-close.directive.ts`: Renamed `bit-dialog-close` to `bitDialogClose`

## Screenshots

image

Screenshot 2023-09-01 at 11 10 58 AM

image

image

image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by Hinton — COMMENTED** [review]
Looks good, any reason this is in draft?

File: apps/browser/src/popup/settings/fingerprint-dialog.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<bit-simple-dialog>
+  <span bitDialogTitle> {{ "yourAccountsFingerprint" | i18n }}: </span>
```
[the remark this answers is not in the document]
Comment (willmartian) [reply apps/browser/src/popup/settings/fingerprint-dialog.component.html]: We do want this one, added a screen shot.

File: apps/desktop/src/services/native-message-handler.service.ts [hunk 2]
```diff
@@ -87,21 +90,12 @@ export class NativeMessageHandlerService {
...
-        showCancelButton: true,
-        cancelButtonText: this.i18nService.t("no"),
-        showConfirmButton: true,
-        confirmButtonText: this.i18nService.t("yes"),
-        allowOutsideClick: false,
-        focusCancel: true,
-      });
 
-      if (submitted.value !== true) {
+      const submitted = await firstValueFrom(
```
Comment (Hinton) [inline apps/desktop/src/services/native-message-handler.service.ts:94]: Should we rename submitted to something more useful?

File: apps/desktop/src/app/components/browser-sync-verification-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    <p>{{ "verifyBrowserDesc" | i18n }}</p>
+    <p>
+      <strong>{{ params.fingerprint.join("-") }}</strong>
+    </p>
+  </span>
+  <ng-container bitDialogFooter>
+    <button bitButton type="submit" buttonType="primary" [bitDialogClose]="true">
+      {{ "approve" | i18n }}
+    </button>
+    <button bitButton type="submit" buttonType="secondary" [bitDialogClose]="false">
```
Comment (Hinton) [inline apps/desktop/src/app/components/browser-sync-verification-dialog.component.html:15]: Two submit buttons?

**Review by shane-melton — DISMISSED** [review]
AC code change looks good to me!

File: apps/browser/src/background/nativeMessaging.background.ts [hunk 4]
```diff
@@ -422,9 +422,10 @@ export class NativeMessagingBackground {
   }
 
   private async showFingerprintDialog() {
-    const fingerprint = (
-      await this.cryptoService.getFingerprint(await this.stateService.getUserId(), this.publicKey)
-    ).join(" ");
+    const fingerprint = await this.cryptoService.getFingerprint(
+      await this.stateService.getUserId(),
+      this.publicKey
+    );
```
[the remark this answers is not in the document]
Comment (willmartian) [reply apps/browser/src/background/nativeMessaging.background.ts:428]: Yes; I moved the formatting of the fingerprint into the dialogs where it is used.

**Review by jlf0dev — DISMISSED** [review]
LGTM 👍

**Review by danielleflinn — APPROVED** [review]

**Review by Hinton — APPROVED** [review]

**Comment by willmartian** [comment]
[cfc17f0](https://github.com/bitwarden/clients/pull/6138/commits/cfc17f0c50969abaeea086aacbccb47eddf336b0) fixes an issue where the browser integration was showing an error dialog on top of the awaiting desktop verification dialog. SweetAlert2 only supported one dialog open at a time, but the CL dialogs don't have this restriction and we must manually close the prior dialog.

**Review by shane-melton — DISMISSED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LTGM!

File: libs/components/src/dialog/directives/dialog-close.directive.ts [hunk 5]
```diff
@@ -5,9 +5,9 @@ import { Directive, HostListener, Input, Optional } from "@angular/core";
   selector: "[bitDialogClose]",
 })
 export class DialogCloseDirective {
-  @Input("bit-dialog-close") dialogResult: any;
+  @Input("bitDialogClose") dialogResult: any;
 
-  constructor(@Optional() public dialogRef: DialogRef<any>) {}
```
Comment (justindbaur) [inline libs/components/src/dialog/directives/dialog-close.directive.ts:10]: I always love the deletion of an `any`!

**Review by justindbaur — APPROVED** [review]
Great stuff!

**Review by Hinton — DISMISSED** [review]

File: apps/desktop/src/services/native-messaging.service.ts [hunk 6]
```diff
@@ -46,125 +49,121 @@ export class NativeMessagingService {
...
-    // Request to setup secure encryption
-    if ("command" in rawMessage && rawMessage.command === "setupEncryption") {
-      const remotePublicKey = Utils.fromB64ToArray(rawMessage.publicKey);
-
-      // Validate the UserId to ensure we are logged into the same account.
-      const accounts = await firstValueFrom(this.stateService.accounts$);
-      const userIds = Object.keys(accounts);
-      if (!userIds.includes(rawMessage.userId)) {
-        ipcRenderer.send("nativeMessagingReply", { command: "wrongUserId", appId: appId });
+    this.ngZone.run(async () => {
```
Comment (Hinton) [inline apps/desktop/src/services/native-messaging.service.ts:52]: This seems a bit odd, is there anything here that needs Angular?

Comment (Hinton) [reply apps/desktop/src/services/native-messaging.service.ts:52]: Shouldn't it be enough to just wrap the dialog portion in ngZone?

**Review by Hinton — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]
LGTM!

**Review by shane-melton — DISMISSED** [review]

**Review by Hinton — APPROVED** [review]

**Review by shane-melton — APPROVED** [review]

**Review by jlf0dev — APPROVED** [review]

**Pull request merged**